script_test.go

```diff
@@ -1768,6 +1768,52 @@ func TestWithStdout_SetsSpecifiedWriterAsStdout(t *testing.T) {
 	}
 }
 
+func TestWithEnv_UnsetsAllEnvVarsGivenEmptySlice(t *testing.T) {
+	t.Parallel()
+	want := []string{}
+
+	output, err := script.NewPipe().WithEnv(want).Exec("printenv").String()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if output != "" {
+		t.Errorf("want empty environment, got %q", output)
+	}
+}
+
+func TestWithEnvChangesExecutionEnvironment(t *testing.T) {
+	t.Parallel()
+	env := []string{"ENV1=test1", "ENV2=test2"}
+
+	got, err := script.NewPipe().WithEnv(env).Exec("printenv").String()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := "ENV1=test1\nENV2=test2\n"
+
+	if got != want {
+		t.Errorf("want %v, got %v", want, got)
+	}
+}
+
+func TestNotSettingEnvFallsBackToDefaultEnvironment(t *testing.T) {
```

Comment:
I don't honestly think we need this test (and that's not something I often find myself saying!)

After all, it's essentially saying that if you don't use `WithEnv`, nothing special happens—and that was already the case.